### PR TITLE
move to rust 2018

### PR DESCRIPTION
**Pull request by antoine-de**
There are pending PR, and they likely will be in conflict with those changes, but since they have not been active for a while I think it's a good time to move to rust 2018.

### CODE REVIEW

File: src/osm_reader/poi.rs [hunk 1]
```diff
@@ -68,9 +68,9 @@ impl Default for PoiConfig {
     }
 }
 impl PoiConfig {
-    pub fn from_reader<R: io::Read>(r: R) -> Result<PoiConfig, Box<Error>> {
-        let mut res: PoiConfig = try!(serde_json::from_reader(r));
-        try!(res.check());
+    pub fn from_reader<R: io::Read>(r: R) -> Result<PoiConfig, Box<dyn Error>> {
+        let mut res: PoiConfig = r#try!(serde_json::from_reader(r));
```
[the remark this answers is not in the document]
Comment (antoine-de) [reply src/osm_reader/poi.rs]: I tried [rerast](https://github.com/google/rerast/blob/master/COOKBOOK.md#replace-trysomething-with-something) but I did not manage to use it so I had to update them manually :roll_eyes:

File: src/admin_geofinder.rs [hunk 2]
```diff
@@ -140,7 +140,7 @@ impl AdminGeoFinder {
     }
 
     /// Iterates on all the admins with a not None boundary.
-    pub fn admins<'a>(&'a self) -> Box<Iterator<Item = Admin> + 'a> {
+    pub fn admins<'a>(&'a self) -> Box<dyn Iterator<Item = Admin> + 'a> {
```
Comment (TeXitoi) [inline src/admin_geofinder.rs]: Should use `impl Iterator<Item = Admin>`

File: src/admin_geofinder.rs [hunk 3]
```diff
@@ -159,7 +159,7 @@ impl AdminGeoFinder {
     }
 
     /// Iterates on all the `Rc<Admin>` in the structure as returned by `get`.
-    pub fn admins_without_boundary<'a>(&'a self) -> Box<Iterator<Item = Arc<Admin>> + 'a> {
+    pub fn admins_without_boundary<'a>(&'a self) -> Box<dyn Iterator<Item = Arc<Admin>> + 'a> {
```
Comment (TeXitoi) [inline src/admin_geofinder.rs]: Should use `impl Iterator`

File: tests/tests.rs [hunk 4]
```diff
@@ -148,7 +148,7 @@ impl<'a> ElasticSearchWrapper<'a> {
         &self,
         word: &str,
         predicate: F,
-    ) -> Box<Iterator<Item = mimir::Place> + 'b>
+    ) -> Box<dyn Iterator<Item = mimir::Place> + 'b>
```
Comment (TeXitoi) [inline tests/tests.rs]: `impl Iterator`

File: tests/tests.rs [hunk 5]
```diff
@@ -159,7 +159,7 @@ impl<'a> ElasticSearchWrapper<'a> {
...
```
Comment (TeXitoi) [inline tests/tests.rs]: `impl Iterator`

File: tests/tests.rs [hunk 6]
```diff
@@ -171,7 +171,7 @@ impl<'a> ElasticSearchWrapper<'a> {
         word: &str,
         predicate: F,
         search_on_global_stops: bool,
-    ) -> Box<Iterator<Item = mimir::Place> + 'b>
+    ) -> Box<dyn Iterator<Item = mimir::Place> + 'b>
```
Comment (TeXitoi) [inline tests/tests.rs]: `impl Iterator`

File: tests/tests.rs [hunk 7]
```diff
@@ -221,19 +221,19 @@ impl<'a> ElasticSearchWrapper<'a> {
...
                                 .filter(predicate),
                         )
-                            as Box<Iterator<Item = mimir::Place>>)
+                            as Box<dyn Iterator<Item = mimir::Place>>)
                     }
                     _ => None,
                 }
             })
-            .unwrap_or(Box::new(None.into_iter()) as Box<Iterator<Item = mimir::Place>>)
+            .unwrap_or(Box::new(None.into_iter()) as Box<dyn Iterator<Item = mimir::Place>>)
```
Comment (TeXitoi) [inline tests/tests.rs]: Some tricks may be needed here to switch to `impl Iterator`

**Pull request merged**